x/incentive/types/params.go

```diff
@@ -0,0 +1,122 @@
+package types
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	cdptypes "github.com/kava-labs/kava/x/cdp/types"
+	kavadistTypes "github.com/kava-labs/kava/x/kavadist/types"
+	tmtime "github.com/tendermint/tendermint/types/time"
+)
+
+// Parameter keys and default values
+var (
+	KeyActive                = []byte("Active")
+	KeyRewards               = []byte("Rewards")
+	DefaultActive            = false
+	DefaultRewards           = Rewards{}
+	DefaultPreviousBlockTime = tmtime.Canonical(time.Unix(0, 0))
+	GovDenom                 = cdptypes.DefaultGovDenom
+	PrincipalDenom           = "usdx"
+	IncentiveMacc            = kavadistTypes.ModuleName
+)
+
+// Params governance parameters for the incentive module
+type Params struct {
+	Active  bool    `json:"active" yaml:"actlive"` // top level governance switch to disable all rewards
+	Rewards Rewards `json:"rewards_periods" yaml:"rewards_periods"`
+}
+
+// NewParams returns a new params object
+func NewParams(active bool, periods Rewards) Params {
+	return Params{
+		Active:  active,
+		Rewards: periods,
+	}
+}
+
+// DefaultParams returns default params for kavadist module
+func DefaultParams() Params {
+	return NewParams(DefaultActive, DefaultRewards)
+}
+
+// String implements fmt.Stringer
+func (p Params) String() string {
+	return fmt.Sprintf(`Params:
+	Active: %t
+	Rewards Periods %s`, p.Active, p.Rewards)
+}
+
+// ParamKeyTable Key declaration for parameters
+func ParamKeyTable() params.KeyTable {
+	return params.NewKeyTable().RegisterParamSet(&Params{})
+}
+
+// ParamSetPairs implements the ParamSet interface and returns all the key/value pairs
+func (p *Params) ParamSetPairs() params.ParamSetPairs {
+	return params.ParamSetPairs{
+		{Key: KeyActive, Value: &p.Active},
+		{Key: KeyRewards, Value: &p.Rewards},
+	}
+}
+
+// Validate checks that the parameters have valid values.
+func (p Params) Validate() error {
+	rewardDenoms := make(map[string]bool)
+
+	for _, reward := range p.Rewards {
+		if rewardDenoms[reward.Denom] {
+			return fmt.Errorf("cannot have duplicate reward denoms: %s", reward.Denom)
+		}
+		rewardDenoms[reward.Denom] = true
+		if !reward.Reward.IsPositive() {
+			return fmt.Errorf("reward must be positive, is %s for %s", reward.Reward, reward.Denom)
+		}
+		if int(reward.Duration.Seconds()) < 0 {
```

Comment:
Any reason we should allow `Duration` to be 0? Same with `TimeLock` on line 80.
```suggestion
		if int(reward.Duration.Seconds()) <= 0 {
```

Reply:
Agree on the first one. The timelock being zero is fine, just means that rewards are immediately liquid.